src/mame/drivers/renegade.cpp

```diff
@@ -234,9 +251,13 @@ TIMER_DEVICE_CALLBACK_MEMBER(renegade_state::interrupt)
 {
 	int scanline = param;
 
-	// vblank irq?
-	if (scanline == 240)
+	// nmi  8 lines before vsync
+	if (scanline == 265)
 		m_maincpu->set_input_line(INPUT_LINE_NMI, ASSERT_LINE);
+	
+	// irq  16 clks per frame
+	if ((scanline % 18) == 0)
```

Comment:
Does it really count to 17? Or does it do 1 irq every 16 lines, and inhibits irqs at let's say vblank?

Reply:
There's 16 clocks per frame but unevenly spaced, the pulses are stretched a bit before/after vsync, [see here](https://photos.app.goo.gl/wj6cwkJqzFC6bEaA9). The top trace is vsync, bottom is the signal to the irq 74ls74a flipflop clk pin (so +ve edge). It fires on lines 16,40,56,72,88,104,120,136,152,168,184,200,216,232,248,264, with this code I get lines 0,18,36,54,72,90,108,126,144,162,180,198,216,234,252,270, which I thought was close enough considering all the game does is read the 3 user inputs (coin1,2,service). It's just using it as a timer to poll the inputs every ~1ms, the only strict timing requirement is not to clash with the nmi at line 265. Although it would be better to be exact I guess...

Reply:
Good measurement! yes, it's better to be accurate, even if the result is not noticeable.
So, I'd add a comment clarifying it, something like: // 16 IRQs per frame: 1 IRQ every 16 lines, but increased to 24 lines in vblank

and then (please double check this equation, or maybe you can make a cleaner one)
if (scanline == 0x10 || (scanline > 0x20 && (scanline & 0xf) == 8))

Reply:
Spot on! thanks